src/form_component.js

```diff
@@ -0,0 +1,153 @@
+import React from 'react';
+
+function Checkinput(props){
+	return(
+		<div className={"inputbox"}>
+			<label className={"inputL"}>
+				{props.title}<input type={props.type} onChange={props.checkformat} className={"inputC"} />
+			</label>
+			<span style={props.permitted? {color:"green"} : {color:"red"}}>{props.format}</span>
+		</div>
+	)
+}
+function Show(props){
+	return(
+		<div style={props.submitted? {} : {display:"none"}}>
+			<h3>User name:{props.name}</h3>
+			<h3>Password:{props.psw}</h3>
+			<h3>Phone:{props.phone}</h3>
+		</div>
+	)
+}
+class RegistrationForm extends React.Component{
+	timer;
+	constructor(props){
+		super();
+		this.state={
+			name:"",
+			psw:"",
+			phone:"",
+			permittedname:false,
+			permittedpsw:false,
+			confirmpsw:false,
+			permittedphone:false,
+			submitted:false,
+		}
+
+		this.handleSubmit = this.handleSubmit.bind(this);
+		this.checkName = this.checkName.bind(this);
+		this.checkPsw = this.checkPsw.bind(this);
+		this.confirmPsw = this.confirmPsw.bind(this);
+		this.checkPhone = this.checkPhone.bind(this);
+	}
+	checkName(e){
+		console.log("checked name");
+		const namestr = e.target.value;
+		clearTimeout(this.timer);
+		this.timer = setTimeout((namestr) => {
+			const name = this.delHtmlTag( namestr );
+			if(name !== ""){
+				this.setState({
+					name: name,
+					permittedname: true,
+				})
+			}else{
+				this.setState({
+					name: name,
+					permittedname:false,
+				})
+			}
+		} ,1000 ,namestr);
+		
+	}
+	checkPsw(e){
+		const pswstr = e.target.value;
+		clearTimeout(this.timer);
+		this.timer = setTimeout((pswstr)=>{
+			const psw = this.delHtmlTag( pswstr );
+			const regUpper = /[A-Z]/;
+			const reglower = /[a-z]/;
+			if(regUpper.test(psw) && reglower.test(psw) && psw.length >= 8){
+				this.setState({
+					psw: psw,
+					permittedpsw: true,
+				})
+			}else{
+				this.setState({
+					psw: psw,
+					permittedpsw:false,
+				})
+			}
+		} ,1000 ,pswstr);
+	}
+	confirmPsw(e){
+		const pswstr = e.target.value;
+		clearTimeout(this.timer);
+		this.timer = setTimeout((pswstr)=>{
+			const psw = this.delHtmlTag( pswstr);
+			if(psw === this.state.psw){
+				this.setState({
+					confirmpsw: true,
+				})
+			}else{
+				this.setState({
+					confirmpsw:false,
+				})
+			}
+		} ,1000 ,pswstr);
+	}
+	checkPhone(e){
+		const phonestr = e.target.value;
+		clearTimeout(this.timer);
+		this.timer = setTimeout((phonestr)=>{
+			const phone = this.delHtmlTag( phonestr );
+			if(phone.match(/^1(3|4|5|7|8)\d{9}$/) || phone.match(/^(\(\d{3,4}\)|\d{3,4}-|\s)?\d{7,14}$/)){
+				this.setState({
+					phone: phone,
+					permittedphone: true,
+				})
+			}else{
+				this.setState({
+					phone: phone,
+					permittedphone:false,
+				})
+			}
+			//console.log(this.state.phone);
+			//console.log(this.state.permittedphone);
+		} ,1000 ,phonestr);
+	}
+
+	handleSubmit(e){
+		this.setState({
+			submitted: true,
```

Comment:
this state also need to set false when one of input is changed. OR following case will happen:
1. input valid data into all inputs, and make "Submit" clickable.
2. click "Submit".
3. User info will show.
4. change any inputs, although the input is invalid, and "submit" button is disabled. The new input will be shown out.